xmtp_proto/src/types/group_message.rs

```diff
@@ -0,0 +1,71 @@
+use super::{Cursor, GroupId};
+use crate::ConversionError;
+use chrono::Local;
+use derive_builder::Builder;
+use openmls::prelude::ContentType;
+
+/// A GroupMessage from the network
+#[derive(Clone, Builder)]
+#[builder(setter(into), build_fn(error = "ConversionError"))]
+pub struct GroupMessage {
```

Comment:
these are the types tha xmtp_mls will use instead of proto

decoupling mls from proto could be useful to avoid large refactors like this. we can handle versioning in the API to keep it away from the MLS code, where it feels like we want to be more concerned with performance, safety and security